gnd/src/main/java/com/google/android/gnd/repository/OfflineBaseMapRepository.java

```diff
@@ -186,4 +212,23 @@ public Completable addAreaAndEnqueue(LatLngBounds bounds) {
                     .filter(tileSource -> tileSource.getState() == TileSource.State.DOWNLOADED)
                     .collect(toImmutableSet()));
   }
+
+  /**
+   * Delete an offline base map and any tile sources associated with it that do not overlap with
+   * other offline base maps .
+   */
+  public Completable deleteArea(String offlineAreaId) {
+    return localDataStore
+        .getOfflineAreaById(offlineAreaId)
+        .flatMapMaybe(this::getIntersectingDownloadedTileSourcesOnce)
+        .flatMapPublisher(Flowable::fromIterable)
+        .map(
+            tileSource ->
+                tileSource.toBuilder().setAreaCount(tileSource.getAreaCount() - 1).build())
+        .flatMapCompletable(
```

Comment:
Also, if the previous stream is a `Completable`, here we could just use `andThen`.

Reply:
done